daemon/cmd/daemon.go

```diff
@@ -693,10 +676,13 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup, params *daemonParams
 		log.WithError(err).Error("error while opening/creating BPF maps")
 		return nil, nil, fmt.Errorf("error while opening/creating BPF maps: %w", err)
 	}
-	// Upsert restored CIDRs after the new ipcache has been opened above
-	if len(restoredCIDRidentities) > 0 {
-		d.ipcache.UpsertGeneratedIdentities(restoredCIDRidentities, nil)
+
+	// Upsert restored CIDRs and their identities in to the metadata layer
+	// Note that the identities themselves were reserved in the local allocator above
+	for prefix, nid := range d.restoredCIDRs {
+		d.ipcache.UpsertMetadata(prefix, source.Restored, restoredCIDRResource, ipcachetypes.RestoredIdentity(nid))
```

Comment:
nit: rather than `restoredCIDRResource` , we may want to do something like `ipcachetypes.NewResourceID(ipcachetypes.ResourceKindDaemon, "", "restored")`

^ idea being to try to (a) reuse the same symbols for easier grep and (b) uniquely identify callsites. Usually `ResourceID` is about the k8s resource, but for in-daemon resources I think we can abuse the last field.

(I noticed this will collide with the following calls)

```
$ git grep ResourceKindDaemon
daemon/cmd/daemon.go:                   ipcachetypes.NewResourceID(ipcachetypes.ResourceKindDaemon, "", ""),
daemon/cmd/datapath.go: daemonResourceID := ipcachetypes.NewResourceID(ipcachetypes.ResourceKindDaemon, "", "")
pkg/ipcache/types/types.go:     ResourceKindDaemon   = ResourceKind("daemon")
```

At some point these should be exposed through API / sysdump so we can properly validate how ipcache entries are formed.

Reply:
Actually this isn't a nit. If we have multiple codepaths using the same resource ID then they can delete entries that were inserted by a completely different codepath.

Fix: https://github.com/cilium/cilium/pull/28770

Reply:
I guess technically right now it's fine because the data types are not the same, but I feel like it's reasonable to split these out further by resourceID.